File: packages/models/dart/view_interop.dart
import 'dart:js_interop';

@JS('initialViewData')
external InitialViewData get initialViewData;

class InitialViewData {
  final String appIdentifier;
  final int randomIntValue;
  
  InitialViewData({required this.appIdentifier, required this.randomIntValue});
}

extension type Time._(JSObject _) implements JSObject {
  external Time(int hours, int minutes);
  external factory Time.onlyHours(int hours);

  external static Time dinnerTime;
  external static Time getTimeDifference(Time t1, Time t2);

  external int hours;
  external int minutes;
  external bool isDinnerTime();

  bool isMidnight() => hours == 0 && minutes == 0;
}

File: packages/models/javascript/viewInterop.js
export class InitialViewData {
  constructor(appIdentifier, randomIntValue) {
    this.appIdentifier = appIdentifier;
    this.randomIntValue = randomIntValue;
  }
  get appIdentifier() {
    return this._appIdentifier;
  }

  set appIdentifier(value) {
    this._appIdentifier = value;
  }

  get randomIntValue() {
    return this._randomIntValue;
  }

  set randomIntValue(value) {
    this._randomIntValue = value;
  }
}

// Need to expose the type to the global scope.
globalThis.InitialViewData = InitialViewData;
